vllm-tt-metal-llama3-70b/src/logging_utils.py

```diff
@@ -0,0 +1,120 @@
+import os
+import datetime
+import json
+
+from vllm.engine.metrics_types import StatLoggerBase, Stats, SupportsMetricsInfo
+from vllm.engine.metrics import logger
+from vllm.engine.llm_engine import LLMEngine
+
+
+# new init function for LLMEngine to be used in vllm api server (online inference) when init in MQLLMEngine
+original_init = LLMEngine.__init__
+
+
+def init_wrapper(self, *args, **kwargs):
+    original_init(self, *args, **kwargs)  # Call the original __init__
+    num_scheduler_steps = self.scheduler_config.num_scheduler_steps
+    batch_size = self.scheduler_config.max_num_seqs
+    self.stat_loggers["raw_logging"] = RawStatLogger(num_scheduler_steps, batch_size)
+
+
+class RawStatLogger(StatLoggerBase):
+    def __init__(self, num_scheduler_steps, batch_size) -> None:
+        self.time_to_first_token = []
+        self.time_per_output_token = []
+        self.num_scheduler_steps = num_scheduler_steps
+        self.batch_size = batch_size
+        timestamp = datetime.now().strftime("%Y-%m-%d_%H%M%S")
+        self.filepath = f"/home/user/tests/statistics_{timestamp}.jsonl"
```

Comment:
The `/home/user` is the path in the docker container. If a user is following the instructions for running the mock model, an instruction in the README is given to mount the test directory to this location. Should I do this differently?

Reply:
No that's perfect -- thanks for the explanation.

Reply:
Actually, I made a change to store it in `cache_root` since this path is the persistent storage rather than have it go under the `tests` directory. I think this helps it be more clean and is more general so if someone runs the actual model, the logging is not saved to the `test` directory.